src/components/cards/SingleCard.jsx

```diff
@@ -3,13 +3,19 @@ import { useState } from "react";
 import { NavLink } from "react-router-dom";
 import { getIcon } from "functions";
 import { useDeleteFunction } from "functions";
+import { useLike } from "context/like-context";
+import { useWatchLater } from "context/watchlater-context";
+import { usePlaylist } from "context/playlist-context";
 
 export const SingleCard = ({ item, flag }) => {
   const [showMenu, setShowMenu] = useState(false);
   const getFunctionName = useDeleteFunction();
   const clickHandler = () => {
     setShowMenu(!showMenu);
```

Comment:
Suggestion you should have wrote a call back only where there is the click handler and also you should have used callback inside the setShowMenu

Reply:
That's a click handler and I wanted to write the CB function separate, And for callback in setShowMenu I will do that